### DISCUSSION TITLE
GitHub Pages InvalidDNSError

### DISCUSSION THREAD

**Discussion by loeshat** [community/community#92761] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi, I'm currently getting this error when I try to enter a new custom domain name. The custom domain name is: www.leoshi.me
leoshi.me is improperly configured
Domain's DNS record could not be retrieved. For more information, see [documentation](https://docs.github.com/articles/setting-up-a-custom-domain-with-github-pages/) (InvalidDNSError).

I believe I've configured to DNS correctly (i've screenshotted the image below)
![image](https://github.com/community/community/assets/90491848/962ea873-3093-487c-98d5-273329a6e8ba)

I've also included a screenshot of when I run the dig command in a terminal:
<img width="492" alt="image" src="https://github.com/community/community/assets/90491848/8bc79f15-825b-4c1d-a4fe-78eed70dd0c9">

Could someone please help me decipher where I went wrong? Thanks!

**Comment by JerryI** [comment]
Just for the others. It takes time for DNS servers to updated cache. So just wait ;)

**Comment by MohamedShiras** [comment]
This is likely just a DNS propagation delay, which is totally normal. Depending on where you purchased your domain, it can take between 12 to 48 hours for the DNS records to fully update. I recommend waiting a few hours and trying again.

**Comment by duyxyz** [comment]
Tung tung tung shahur

**Comment by rohitks12987** [comment]
The error **“Domain's DNS record could not be retrieved (InvalidDNSError)”** occurs when the DNS configuration for the custom domain is not correctly recognized. This usually happens when the required DNS records are missing, incorrect, or have not fully propagated.

To resolve this issue, the DNS settings must be properly configured. For the **www subdomain ([[www.leoshi.me](http://www.leoshi.me/)](http://www.leoshi.me))**, a **CNAME record** should be added in the DNS configuration that points to the GitHub Pages site (for example, `username.github.io`). If the root domain **leoshi.me** is also used, it should contain the four GitHub Pages **A records**: `185.199.108.153`, `185.199.109.153`, `185.199.110.153`, and `185.199.111.153`.

Additionally, the repository should contain a **CNAME file** in the root directory with the custom domain name (`www.leoshi.me`). After making these changes, it may take some time for DNS propagation (typically a few minutes up to 24 hours). Once the DNS records propagate successfully, the domain should work correctly with GitHub Pages and the error will be resolved.

**Reply by JerryI** [reply]
Stop using LLMs for answers!!!
